src/python/WMCore/MicroService/MSUnmerged/MSUnmerged.py

```diff
@@ -771,6 +771,36 @@ def uploadRSEToMongoDB(self, rse, fullRSEToDB=False, overwrite=True):
             raise MSUnmergedPlineExit(msg)
         return rse
 
+    def getStatsFromMongoDB(self, detail=False, **kwargs):
+        """
+        Auxiliary method to serve the APIs of the REST calls for the service.
+        Implements various queries to MongoDB based on the parameters passed.
+        :param detail:           Bool marking if additional details must be queried from the database
+                                 (e.g. putting the 'dirs' field into the mongo projection)
+        :param rse=rseName:      String representing the RSE name to query the database for
+        :return:                 Dictionary with the type of query + the data returned from the database
+        """
+        data = {}
+        if kwargs.get('rse'):
+            data["query"] = 'rse=%s&detail=%s' % (kwargs['rse'], detail)
+            mongoProjection = {"_id": False,
+                               "name": True,
+                               "pfnPrefix": True,
+                               "isClean": True,
+                               "timestamps": True,
+                               "counters": True}
+            if detail:
+                mongoProjection["dirs"]  = True
+
+            rseList = kwargs['rse'] if isinstance(kwargs['rse'], list) else [kwargs['rse']]
+            data["rseData"] = []
+            for rseName in rseList:
+                mongoFilter = {'name': rseName}
+                data["rseData"].append(self.msUnmergedColl.find_one(mongoFilter, projection=mongoProjection))
+        if kwargs.get('rse_type'):
```

Comment:
If we have a call like: `rse=Alan&rse_type=Blah&detail=False`
it would result in:
```
data["query"] = "rse_type=Blah"
```
dropping the previous content, while what we should do is actually to concatenate them through a `&` char.

Reply:
Sorry @amaltaro I tend to disagree here. Those two interfaces simply cannot live in a single call to thee database (or at least not in a simple one). Those are serving completely different purposes. So mixture of the sort simply makes things unneeded over complicated. The rest call `rse_type=....` would result (eventually in the future) in an aggregation across all the RSEs of the given type through out the whole database and providing results based on completely different projection than the one for the `rse=....` call. On the other hand the call `rse=...` is simply fetching a single document (eventually could be repeated multiple times and the result displayed in a list) and displaying it according to the projection we already discussed above. So this mixture of calls if not completely impossible in theory is at least not going to produce results of similar type and shape, so I'd not support these two calls together.  I'd rather instruct the users that  the better way to fetch those tow difernt types of info simply by making two separate REST calls to the relevant API.

Reply:
BTW this:
> If we have a call like: rse=Alan&rse_type=Blah&detail=False
> it would result in:
>
> data["query"] = "rse_type=Blah"

Is completely true, and I kind of like this behavior more (which I already explained why). And having the interfaces  work separately was kind of a desired effect rather than a defect.